includes/class-products.php

```diff
@@ -95,103 +107,180 @@ public static function is_active() {
-			$featured_upgrade_single->set_name( __( '“Featured” Listing Upgrade', 'newspack-listings' ) );
-			$featured_upgrade_single->set_regular_price( $settings[ self::PRODUCT_META_KEYS['featured'] ] );
-			$featured_upgrade_single->update_meta_data( '_newspack_listings_product_slug', self::PRODUCT_META_KEYS['featured'] );
-			$featured_upgrade_single->set_virtual( true );
-			$featured_upgrade_single->set_downloadable( true );
-			$featured_upgrade_single->set_catalog_visibility( 'hidden' );
-			$featured_upgrade_single->set_sold_individually( true );
-			$featured_upgrade_single->save();
-
-			// Monthly subscription product.
-			$monthly_product = new \WC_Product_Subscription();
-			/* translators: %s: Product name */
-			$monthly_product->set_name( __( 'Listing Subscription', 'newspack-listings' ) );
-			$monthly_product->set_regular_price( $settings[ self::PRODUCT_META_KEYS['subscription'] ] );
-			$monthly_product->update_meta_data( '_newspack_listings_product_slug', self::PRODUCT_META_KEYS['subscription'] );
-			$monthly_product->update_meta_data( '_subscription_price', wc_format_decimal( $settings[ self::PRODUCT_META_KEYS['subscription'] ] ) );
-			$monthly_product->update_meta_data( '_subscription_period', 'month' );
-			$monthly_product->update_meta_data( '_subscription_period_interval', 1 );
-			$monthly_product->set_virtual( true );
-			$monthly_product->set_downloadable( true );
-			$monthly_product->set_catalog_visibility( 'hidden' );
-			$monthly_product->set_sold_individually( true );
-			$monthly_product->save();
-
-			// Monthly "premium subscription" upgrade.
-			$premium_upgrade_monthly = new \WC_Product_Subscription();
-			/* translators: %s: Product name */
-			$premium_upgrade_monthly->set_name( __( 'Premium Subscription Upgrade', 'newspack-listings' ) );
-			$premium_upgrade_monthly->set_regular_price( $settings[ self::PRODUCT_META_KEYS['premium'] ] );
-			$premium_upgrade_monthly->update_meta_data( '_newspack_listings_product_slug', self::PRODUCT_META_KEYS['premium'] );
-			$premium_upgrade_monthly->update_meta_data( '_subscription_price', wc_format_decimal( $settings[ self::PRODUCT_META_KEYS['premium'] ] ) );
-			$premium_upgrade_monthly->update_meta_data( '_subscription_period', 'month' );
-			$premium_upgrade_monthly->update_meta_data( '_subscription_period_interval', 1 );
-			$premium_upgrade_monthly->set_virtual( true );
-			$premium_upgrade_monthly->set_downloadable( true );
-			$premium_upgrade_monthly->set_catalog_visibility( 'hidden' );
-			$premium_upgrade_monthly->set_sold_individually( true );
-			$premium_upgrade_monthly->save();
-
-			$parent_product->set_children(
-				[
-					$single_product->get_id(),
-					$featured_upgrade_single->get_id(),
-					$monthly_product->get_id(),
-					$premium_upgrade_monthly->get_id(),
-				]
-			);
-			$parent_product->save();
-			update_option( self::PRODUCT_OPTION, $parent_product->get_id() );
+	/**
+	 * Delete self-serve listing products.
```

Comment:
Some of the products are subscriptions. If they are deleted, the subscriptions will be presumably cancelled. Maybe not in this PR, but shouldn't there be warning if there are any active subscriptions?

Reply:
This is a good idea, but since this is an MVP hopefully a simple warning explanation next to the link will suffice: 97c40cf